### PR TITLE
An alternate solution for pairs_with_given_sum

### PR DESCRIPTION
**Pull request by PavanVenkataNagaManoj17**
### Describe your change:
implemented a very simple alternative solution instead of using the itertools package

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: data_structures/arrays/pairs_with_given_sum.py [hunk 1]
```diff
@@ -23,6 +23,19 @@ def pairs_with_sum(arr: list, req_sum: int) -> int:
...
+def no_of_pairs(arr: list, req_sum: int) -> int:
+    # this is a non optimised code and easy to understand as i am a beginner in coding
+    # the code above is the best one to use as this function uses nested loops which is
+    # very much slower in execution
+    count_total = 0
+    for i in range(len(arr)):
+        for j in range(i + 1, len(arr)):
+            if (arr[i] + arr[j]) == req_sum:
+                count_total += 1
+    return count_total
```
Comment (imSanko) [inline data_structures/arrays/pairs_with_given_sum.py:36]:
```suggestion
def no_of_pairs(arr: list, req_sum: int) -> int:
    count_total = 0
    seen = {}
    
    for number in arr:
        complement = req_sum - number
        count_total += seen.get(complement, 0)
        seen[number] = seen.get(number, 0) + 1
    
    return count_total

```

**Review by imSanko — CHANGES_REQUESTED** [review]

File: data_structures/arrays/pairs_with_given_sum.py [hunk 2]
```diff
@@ -23,6 +23,19 @@ def pairs_with_sum(arr: list, req_sum: int) -> int:
...
 
+# Without using the  itertools Package we can do it too
+def no_of_pairs(arr: list, req_sum: int) -> int:
+    # this is a non optimised code and easy to understand as i am a beginner in coding
+    # the code above is the best one to use as this function uses nested loops which is
+    # very much slower in execution
+    count_total = 0
+    for i in range(len(arr)):
+        for j in range(i + 1, len(arr)):
+            if (arr[i] + arr[j]) == req_sum:
```
Comment (tianyizheng02) [inline data_structures/arrays/pairs_with_given_sum.py:34]:
```suggestion
            if arr[i] + arr[j] == req_sum:
```
The parentheses are unnecessary

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

File: data_structures/arrays/pairs_with_given_sum.py [hunk 3]
```diff
@@ -23,6 +23,19 @@ def pairs_with_sum(arr: list, req_sum: int) -> int:
     return len([1 for a, b in combinations(arr, 2) if a + b == req_sum])
 
 
+# Without using the  itertools Package we can do it too
+def no_of_pairs(arr: list, req_sum: int) -> int:
```
Comment (tianyizheng02) [inline data_structures/arrays/pairs_with_given_sum.py:27]:
```suggestion
def no_of_pairs(arr: list[int], req_sum: int) -> int:
```
Please specify the type of the list's contents

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

File: data_structures/arrays/pairs_with_given_sum.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (PavanVenkataNagaManoj17) [reply data_structures/arrays/pairs_with_given_sum.py:27]: ok sir

File: data_structures/arrays/pairs_with_given_sum.py [hunk 4]
```diff
@@ -23,6 +23,19 @@ def pairs_with_sum(arr: list, req_sum: int) -> int:
...
+    # this is a non optimised code and easy to understand as i am a beginner in coding
+    # the code above is the best one to use as this function uses nested loops which is
+    # very much slower in execution
```
[the remark this answers is not in the document]
Comment (PavanVenkataNagaManoj17) [reply data_structures/arrays/pairs_with_given_sum.py:30]: ok sir

**Comment by cclauss** [comment]
Closing tests_are_failing PRs to prepare for Hacktoberfest 2024

**Pull request closed without merging**